src/main/java/org/elasticsearch/index/gateway/IndexShardGateway.java

```diff
@@ -78,7 +73,10 @@ public IndexShardGateway(ShardId shardId, @IndexSettings Settings indexSettings,
 
         this.waitForMappingUpdatePostRecovery = indexSettings.getAsTime("index.gateway.wait_for_mapping_update_post_recovery", TimeValue.timeValueSeconds(30));
         syncInterval = indexSettings.getAsTime("index.gateway.sync", TimeValue.timeValueSeconds(5));
-        if (syncInterval.millis() > 0) {
+        if (indexShard instanceof ShadowIndexShard) {
```

Comment:
yeah, maybe we should have a utility method somewhere? I think the best alternative would be to change the juice context not to create this as it is not needed, but I wanted to keep the scope of the change. Wondering if we should just keep instanceof and do a bigger rewrite as a different change?